merlin/models/tf/transforms/features.py

```diff
@@ -726,10 +726,13 @@ def _target_column_schemas(self):
                 target_columns[t] = self.schema.select_by_name(t).first
             elif isinstance(t, ColumnSchema):
                 target_columns[t.name] = t
-            elif isinstance(t, Schema):
-                selected_schema = t.select_by_tag(Tags.TARGET)
+            elif isinstance(t, Tags):
+                selected_schema = self.schema.select_by_tag(t)
                 for col_name, col_schema in selected_schema.column_schemas.items():
                     target_columns[col_name] = col_schema
+            elif isinstance(t, Schema):
```

Comment:
Do think we should add the tag `TARGET` to the target-columns here?

One use-case I could think of where that might come in handy is:

```python
loader = mm.Loader(dataset, batch_siz=1000, transform=ToTarget(schema, Tags.ITEM_ID))
schema = loader.schema
model = mm.Model(mm.Encoder(schema, mm.MLPBlock([128, 64]), mm.OutputBlock(schema))
```

Reply:
Or would it be better to have `loader.target_schema`?

Reply:
We add the `TARGET` tag in [compute_output_schema](https://github.com/edknv/models/blob/fix_to_target_1/merlin/models/tf/transforms/features.py#L771) instead, so with the latest changes in `Loader` for computing the output schema (#740), we should be able to do
```
schema = loader.output_schema
```
and the target columns in `loader.output_schema` will have `Tags.TARGET` added. @marcromeyn That should cover your example use case?